bluemira/equilibria/run.py

```diff
@@ -254,108 +417,218 @@ def __init__(
+    params: ParameterFrame
+        Parameter frame with which to perform the problem
+    coilset: CoilSet
+        PF coilset to use in the equilibrium design
+    grid: Grid
+        Grid to use in the equilibrium design
+    current_opt_constraints: Optional[List[OptimisationConstraint]]
+        List of current optimisation constraints for equilibria
+    coil_constraints: Optional[List[OptimisationConstraint]]
+        List of coil current optimisation constraints for all snapshots (including
+        breakdown)
+    equilibrium_constraints: List[OptimisationConstraint]
+        List of magnetic constraints to use for equilibria. Depending on the optimisation
+        problem, these may be used in the objective function or constraints
+    profiles: Profile
+        Plasma profile object to use when solving equilibria
+    breakdown_strategy_cls: Type[BreakdownZoneStrategy]
+        BreakdownZoneStrategy class to use when determining breakdown constraints
+    breakdown_problem_cls: Type[BreakdownCOP]
+        Coilset optimisation problem class for the breakdown phase
+    breakdown_optimiser: Optimiser
+        Optimiser for the breakdown
+    equilibrium_problem_cls: Type[CoilsetOptimisationProblem]
+        Coilset optimisation problem class for the equilibria and current vector
+    equilibrium_optimiser: Optimiser
+        Optimiser for the equilibria and current vector
+    equilibrium_convergence: ConvergenceCriterion
+        Convergence criteria to use when solving equilibria
+    equilibrium_settings: Optional[Dict]
+        Settings for the solution of equilibria
+    position_problem_cls: Type[PulsedNestedPositionCOP]
+        Coilset optimisation problem class for the coil positions
+    position_optimiser: Optimiser
+        Optimiser for the coil positions
+    limiter: Optional[Limiter]
+        Limiter to use when solving equilibria
+    """
 
-            if i == 0:
-                psi_1 = psi_premag
-            else:
-                relaxed = np.isclose(psi_premag, psi_1, rtol=1e-2)
-                psi_1 = psi_premag
-            i += 1
-            if i == i_max:
-                raise EquilibriaError(
-                    "Unable to relax the breakdown optimisation for coil sizes."
-                )
+    def __init__(
+        self,
+        params,
+        coilset: CoilSet,
+        position_mapper: PositionMapper,
+        grid: Grid,
+        current_opt_constraints: Optional[List[OptimisationConstraint]],
+        coil_constraints: Optional[List[OptimisationConstraint]],
+        equilibrium_constraints: List[OptimisationConstraint],
+        profiles: Profile,
+        breakdown_strategy_cls: Type[BreakdownZoneStrategy],
+        breakdown_problem_cls: Type[BreakdownCOP],
+        breakdown_optimiser: Optimiser = Optimiser(
+            "COBYLA", opt_conditions={"max_eval": 5000, "ftol_rel": 1e-10}
+        ),
+        breakdown_settings: Dict = {"B_stray_con_tol": 1e-8, "n_B_stray_points": 20},
```

Comment:
This will run into the mutable defaults issue. Possibly we should have some defaults baked into each problem class and set this to None. The dictionaries are fine in the `breakdown_optimser` because they are wrapped in the `Optimiser` object